src/main/java/de/muenchen/isi/domain/service/AbfrageStatusService.java

```diff
@@ -1,59 +1,274 @@
+     */
+    private StateMachine<StatusAbfrage, StatusAbfrageEvents> build(final UUID id) throws EntityNotFoundException {
+        final InfrastrukturabfrageModel abfrage = this.abfrageService.getInfrastrukturabfrageById(id);
+        final StateMachine<StatusAbfrage, StatusAbfrageEvents> stateMachine = this.stateMachineFactory.getStateMachine(abfrage.getId());
+
+        stateMachine.stopReactively().block();
+
+        stateMachine.getStateMachineAccessor().doWithAllRegions(stateMachineAccess -> {
+
+            // Setzt den Status der Abfrage aus der DB in der StateMachine.
+            stateMachineAccess.resetStateMachineReactively(new DefaultStateMachineContext<>(abfrage.getAbfrage().getStatusAbfrage(), null, null, null)).block();
+
+            // Der Interceptor aktualisiert bei einer Statusänderung den Status der in der DB gespeicherten Entität.
+            stateMachineAccess.addStateMachineInterceptor(new StateMachineInterceptorAdapter<>() {
+
+                /**
+                 * Verhalten siehe {@link StateMachineInterceptor#preTransition(StateContext)}
+                 *
+                 * Die Rückgabe des Wertes null führt in der StateMachine zum Ergebnis {@link StateMachineEventResult.ResultType#DENIED}
+                 */
+                @Override
+                public StateContext<StatusAbfrage, StatusAbfrageEvents> preTransition(final StateContext<StatusAbfrage, StatusAbfrageEvents> stateContext) {
+                    final State<StatusAbfrage, StatusAbfrageEvents> state = stateContext.getTransition().getTarget();
+                    final MessageHeaders messageHeaders = stateContext.getMessageHeaders();
+                    try {
+                        final UUID abfrageId = AbfrageStatusService.this.getAbfrageId(messageHeaders);
+                        final InfrastrukturabfrageModel abfrage = AbfrageStatusService.this.abfrageService.getInfrastrukturabfrageById(abfrageId);
+                        abfrage.getAbfrage().setStatusAbfrage(state.getId());
+                        AbfrageStatusService.this.abfrageService.updateInfrastrukturabfrage(abfrage);
+                    } catch (final Exception exception) {
+                        log.error("Die vom Statuswechsel betroffene Infrastrukturabfrage wurde nicht gefunden.", exception);
+                        return null;
+                    }
+                    return stateContext;
+                }
+
+            });
+
+        });
+
+        stateMachine.startReactively().block();
+        return stateMachine;
+    }
+
+    /**
+     * Leitet eine Statusänderung ein mit dem übergebenem Event
+     * <p>
+     *
+     * @param id           vom Typ {@link UUID} um die Abfrage aus der DB zu holen.
+     * @param event        vom Typ {@link StatusAbfrageEvents} um eine Statusänderung durchzuführen
+     * @param stateMachine vom Typ {@link StateMachine} welches den aktuellen Status plus alle Statis enthält
```

Comment:
Plural von Status ist Status. Stati oder Statusse existiert nicht siehe Duden
```suggestion
     * @param stateMachine vom Typ {@link StateMachine} welche den aktuellen Status plus alle Status enthält
```